src/Umbraco.Core/Persistence/Factories/ContentTypeFactory.cs

```diff
@@ -1,98 +1,144 @@
     {
-        private readonly Guid _nodeObjectType;
+        #region IContentType
 
-        public ContentTypeFactory(Guid nodeObjectType)
+        public IContentType BuildContentTypeEntity(ContentTypeDto dto)
         {
-            _nodeObjectType = nodeObjectType;
+            var contentType = new ContentType(dto.NodeDto.ParentId);
+            BuildCommonEntity(contentType, dto);
+
+            //on initial construction we don't want to have dirty properties tracked
+            // http://issues.umbraco.org/issue/U4-1946
+            contentType.ResetDirtyProperties(false);
+
+            return contentType;
         }
 
-        #region Implementation of IEntityFactory<IContentType,DocumentTypeDto>
+        #endregion
 
-        public IContentType BuildEntity(DocumentTypeDto dto)
+        #region IMediaType
+
+        public IMediaType BuildMediaTypeEntity(ContentTypeDto dto)
         {
-            var contentType = new ContentType(dto.ContentTypeDto.NodeDto.ParentId)
-                                  {
-                                      Id = dto.ContentTypeDto.NodeDto.NodeId,
-                                      Key = dto.ContentTypeDto.NodeDto.UniqueId,
-                                      Alias = dto.ContentTypeDto.Alias,
-                                      Name = dto.ContentTypeDto.NodeDto.Text,
-                                      Icon = dto.ContentTypeDto.Icon,
-                                      Thumbnail = dto.ContentTypeDto.Thumbnail,
-                                      SortOrder = dto.ContentTypeDto.NodeDto.SortOrder,
-                                      Description = dto.ContentTypeDto.Description,
-                                      CreateDate = dto.ContentTypeDto.NodeDto.CreateDate,
-                                      Path = dto.ContentTypeDto.NodeDto.Path,
-                                      Level = dto.ContentTypeDto.NodeDto.Level,
-                                      CreatorId = dto.ContentTypeDto.NodeDto.UserId.Value,
-                                      AllowedAsRoot = dto.ContentTypeDto.AllowAtRoot,
-                                      IsContainer = dto.ContentTypeDto.IsContainer,
-                                      Trashed = dto.ContentTypeDto.NodeDto.Trashed,
-                                      DefaultTemplateId = dto.TemplateNodeId
-                                  };
+            var contentType = new MediaType(dto.NodeDto.ParentId);
+            BuildCommonEntity(contentType, dto);
+
             //on initial construction we don't want to have dirty properties tracked
             // http://issues.umbraco.org/issue/U4-1946
             contentType.ResetDirtyProperties(false);
+
             return contentType;
         }
 
-        public DocumentTypeDto BuildDto(IContentType entity)
+        #endregion
+
+        #region IMemberType
+
+        public IMemberType BuildMemberTypeEntity(ContentTypeDto dto)
+        {
+            throw new NotImplementedException();

```

Comment:
It was in MemberTypeFactory but MediaTypeFactory and MemberTypeFactory have been merged here to reduce the (vast) amount of duplicated code. And it was _not_ implemented in MemberTypeFactory...